server.js

```diff
@@ -23,9 +27,142 @@ app.use(express.json())
 
 // Start defining your routes here
 app.get('/', (req, res) => {
-  res.send('Hello world')
+  const hello = {
+    Welcome: 'Hello! This be a Book API',
+    Routes: [{
+      "/books_________________________": 'Get books.',
+      "/books?title='title name'______": 'Get books with matching titles.',
+      "/books?author='author name'____": 'Get books from specific author.',
+      "/books/id/'number'_____________": 'Get books with matching id.',
+      "/books/isbn/'number'___________": 'Get books by unique ISBN nr.',
+      "/books/pages/'number'__________": 'Get books with certain amount of pages.',
+      "/books/language/eng____________": 'Get books written in english.',
+      "/books/language/en-US__________": 'Get books written in US-english.',
+      "/books/rating/'number'_________": 'Get books by rating, number = integer.',
+      "/endpoints_____________________": 'Get API endpoints.'
+    }]
+    
+  }
+  res.send(hello)
+})
+
+// List endpoints
+app.get ('/endpoints', (req, res) => {
+  res.send(listEndpoints(app))
+}
+)
+
+// route for books API
+app.get('/books', (req, res) => {
+  const { title, author } = req.query
+ 
+  let booksToSend = books
+
+  if (title){
+    booksToSend = booksToSend.filter(
+      (item) => item.title.toLowerCase().indexOf(title.toLowerCase()) !== -1 )
+    }
+  if (author){
+    booksToSend = booksToSend.filter(
+      (item) => item.authors.toLowerCase().indexOf(author.toLowerCase()) !== -1 )
+    }
+
+  res.json({
+    response: booksToSend,
+    success: true
+  })
+})
+
+// Specific book from id //books/id/"nr"
+app.get('/books/id/:id', (req, res) => {
+  const { id } = req.params
+
+  const bookNR = books.find(item => item.bookID === +id)
+
+  if (!bookNR) {
+    res.status(404).send('No book found!')
+  } else {
+    res.json(bookNR)
+  }
+})
+
+// Book by ISBN nr
+app.get('/books/isbn/:isbn', (req, res) => {
+  const { isbn } = req.params
+
+  const isbnNr = books.find(item => item.isbn === +isbn)
+
+  if (!isbnNr) {  
+    res.status(404).send('No book for that ISBN nr!')
+  } else {
+    res.json(isbnNr)
+  }
 })
 
+// Books by number of pages
+app.get('/books/pages/:pages', (req, res) => {
+  const { pages } = req.params
+
+  const pageNr = books.filter(item => item.num_pages === +pages)
+
+  if (pageNr.length === 0) {
+    res.status(404).send('No books with that amount of pages!')
+  } else {
+    res.json(pageNr)
+  }
+})
+
+// books in english //books/eng
```

Comment:
I think you only need one endpoint for the language because you have added a placeholder, '/books/language/:placeholder'. So when we get:
/books/language/eng -> english
/books/language/en-US -> english (us)
/books/language//spa -> spanish

So we can just write for example:
```
const { language } = req.params
const booksByLanguage = books.filter( (item) => item.language_code === language )
res.json(booksByLanguage)
```